packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -2938,3 +2926,128 @@ class _RenderCupertinoDialogActions extends RenderBox
     return defaultHitTestChildren(result, position: position);
   }
 }
+
+// A column layout with two widgets, where the top widget expands vertically as
+// needed, and the bottom widget has a minimum height.
+//
+// Both child widgets stretch horizontally to the parent's maximum width
+// constraint, with vertical space allocated in this priority:
+//
+//  1. The `bottom` widget receives its requested height, up to a
+//     `bottomMinHeight` limit.
+//  2. The `top` widget receives its requested height.
+//  3. The `bottom` widget receives additional height as needed.
+//
+// This mirrors the behavior seen in iOS components like action sheets and
+// alerts.
+//
+// Implementing this layout with simple compositing widgets is challenging
+// because:
+//
+//  * The bottom widget should take more than `bottomMinHeight` if the top
+//    widget is short.
+//  * The bottom widget should take less than `bottomMinHeight` if it is
```

Comment:
maybe `bottomMinHeight` needs a better name, the name contradict the statement.

the name bottomMinHeight implies it is the lowest bound.

Reply:
I've changed all "minHeight"s to "maxHeight"s. Sorry for the confusion!

Reply:
See https://github.com/flutter/flutter/pull/150708#discussion_r1661584519